pkg/config/reload.go

```diff
@@ -13,8 +13,8 @@ import (
 )
 
 // Reload reloads the configuration for the single crio.conf and the drop-in
-// configuration directory.
-func (c *Config) Reload() error {
+// configuration directory. Method takes a function to update PinnedImages as parameter.
+func (c *Config) Reload(updateImageCache func([]string) error) error {
```

Comment:
is it possible to use the `_inject_test.go` pattern where we can override for the tests specifically by calling a hidden setter?

Reply:
I've changed a bit of implementation and used another way of passing `Config.` PTAL